kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/ConfigParser.java

```diff
@@ -94,7 +93,6 @@ public static ObjectMapper createObjectMapper() {
                 .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
                 .configure(DeserializationFeature.FAIL_ON_MISSING_EXTERNAL_TYPE_ID_PROPERTY, false)
                 .configure(JsonParser.Feature.STRICT_DUPLICATE_DETECTION, true)
-                .setSerializationInclusion(JsonInclude.Include.NON_DEFAULT)
```

Comment:
When I tried to test a particular case this setting caused conversion to fail. Converting `InsecureTls(false)` it omitted the `insecure` property when serialized, then when it tried to deduce the type of `{}` it failed. It needs to be `{"insecure" : x }`  so it can do the type deduction. I'm not sure why we have it switched to `NON_DEFAULT` but our serialization usages are test only, I imagine it was switched on to produce more succinct, readable json for humans.

Reply:
hmm, there's tests for fidelity checking it produces the same content as it parsed, sort of reasonable. I've reverted and instead added an explicit annotation to `InsecureTls` and a test for this case.